src/main/kotlin/org/rust/lang/core/resolve2/FacadeResolve.kt

```diff
@@ -279,46 +283,62 @@ private fun <T> RsPossibleMacroCall.resolveToMacroAndThen(
 }
 
 fun RsMetaItem.resolveToProcMacroWithoutPsi(): ProcMacroDefInfo? {
+    val resolvedMacro = resolveToProcMacroWithoutPsiUnchecked(checkIsMacroAttr = true)
+        ?: return null
+    if (resolvedMacro.props.treatAsBuiltinAttr) return null
+    return resolvedMacro
+}
+
+fun RsMetaItem.resolveToProcMacroWithoutPsiUnchecked(checkIsMacroAttr: Boolean): ProcMacroDefInfo? {
```

Comment:
If we find a way to get rid of this method, I will be happy

If we actually don't want to return `Attr` instance if `resolveToProcMacroWithoutPsiUnchecked` returns null in `org.rust.lang.core.psi.ProcMacroAttribute.getProcMacroAttribute`, then we can merge `resolveToProcMacroWithoutPsi` and `resolveToProcMacroWithoutPsiUnchecked`

Reply:
Ok, now you can be happy :)